File: go/bob/bob.go
package bob

import (
	"strings"
	//"unicode"
	"regexp"
)

func Hey(remark string) string {
	// Start with whitespace; 0 or more; End with whitespace
	ifSilent := regexp.MustCompile(`^\s*$`).MatchString(remark)
	// Matches a `?`, end with 0 or more whitespace
	ifQuestion := regexp.MustCompile(`\?\s*$`).MatchString(remark)
	// if original string == uppercase of original string and must contain 1 or more alphabet
	ifYelling := (remark == strings.ToUpper(remark)) && regexp.MustCompile(`[A-Za-z]+`).MatchString(remark)

	switch {
	case ifSilent:
		return "Fine. Be that way!"
	case ifQuestion && ifYelling:
		return "Calm down, I know what I'm doing!"
	case ifYelling:
		return "Whoa, chill out!"
	case ifQuestion:
		return "Sure."
	default:
		return "Whatever."

	}
}

File: python/bob/bob.py
import re 
# using regex

def hey(what):
	# Start with whitespace; 0 or more; End with whitespace
    if re.search("^\s*$", what):
        return "Fine. Be that way!"
    # if entire string contains uppercase and no lowercase
    elif what.isupper():
        return "Whoa, chill out!"
    # matches a ?; ends with 0 or more whitespace
    elif re.search("\?\s*$", what):
        return "Sure."
    else:
        return "Whatever."